org.bigtester.ate.core/src/main/java/org/bigtester/ate/model/project/TestProject.java

```diff
@@ -213,21 +216,24 @@ private void runSuites(String filteringSuiteName, String filteringTestCaseName)
 		}
 		if (testng.getTestListeners().stream().filter(listener->listener instanceof ATEXMLReporter).count()==0) {
 			ATEXMLReporter rng = new ATEXMLReporter();
-			rng.setStackTraceOutputMethod(XMLReporterConfig.STACKTRACE_NONE);
+			rng.setStackTraceOutputMethod(XMLReporterConfig.StackTraceLevels.NONE);
 			testng.addListener(rng);
 		}
 		//TODO Can be optimaized in cucumber run, no need to delete old suite cases in each step if it is belong to current test project
 		List<TestSuite> suites = this.getSuiteList();
-		if (filteringTestCaseName!=null) {
-			suites = suites.stream().filter(suite->suite.getSuiteName().equalsIgnoreCase(filteringSuiteName)).collect(Collectors.toList());
-			for (TestSuite tSuite : suites) {
-				tSuite.setTestCaseList(tSuite.getTestCaseList()
-						.stream()
-						.filter(tcase -> tcase.getTestCaseFilePathName()
-								.contains("/" + filteringTestCaseName +".xml"))
-								.collect(Collectors.toList()));
+		for(CucumberFilter f : filter) {
```

Comment:
create a new String array and storing all test case names defined in
cucumber filters.

and then user the .stream().fiter () to get all test cases, whose
testcaseName is in above new String array.